readthedocs/analytics/utils.py

```diff
@@ -17,15 +17,23 @@
 
 
 def get_client_ip(request):
-    """Gets the real IP based on a request object."""
-    ip_address = request.META.get('REMOTE_ADDR')
+    """
+    Gets the real client's IP address.
+
+    It returns the real IP address of the client based on ``HTTP_X_FORWARDED_FOR``
+    header. If the header is not found, it returns ``None``.
+    """
+
+    x_forwarded_for = request.META.get('HTTP_X_FORWARDED_FOR', '')
 
-    # Get the original IP address (eg. "X-Forwarded-For: client, proxy1, proxy2")
-    x_forwarded_for = request.META.get('HTTP_X_FORWARDED_FOR', '').split(',')[0]
     if x_forwarded_for:
-        ip_address = x_forwarded_for.rsplit(':')[0]
+        # HTTP_X_FORWARDED_FOR can be a comma-separated list of IPs.
+        # The client's IP will be the first one.
+        # (eg. "X-Forwarded-For: client, proxy1, proxy2")
+        real_ip = x_forwarded_for.split(',')[0].strip()
+        return real_ip
 
-    return ip_address
+    return None
```

Comment:
There's two cases here and these changes neglect one of them:

* If `X-Forwarded-For` is present, this function should return the first value from the comma separated list. Arguably, it could validate that it is a real IP address, but that isn't that important. 
* If `X-Forwarded-For` is not present, this function should return the value from `REMOTE_ADDR`. This implementation returns `None`.

This function is used in advertising code for geo-targeting as well as being used for server side analytics (currently used in advertising but in the future might replace Google Analytics JS) but the middleware can be safely removed.

Reply:
@davidfischer 
Thank you for the information.
I have updated the PR.

Will there be any case in which both of these headers are not found? Currently the implementation returns `None` in that case.